src/GraphQLParser/AST/GraphQLObjectValue.cs

```diff
@@ -7,10 +8,58 @@ namespace GraphQLParser.AST;
 /// </summary>
 public class GraphQLObjectValue : GraphQLValue
 {
+    private ROM _value;
+    private Dictionary<string, object?>? _object;
+    private static readonly ReadOnlyDictionary<string, object?> _empty = new(new Dictionary<string, object?>());
+
     /// <inheritdoc/>
     public override ASTNodeKind Kind => ASTNodeKind.ObjectValue;
 
+    /// <summary>
+    /// Object value represented as <see cref="ROM"/>.
+    /// </summary>
+    public ROM Value
+    {
+        get => _value;
+        set
+        {
+            _value = value;
+            _object = null;
+        }
+    }
+
+    /// <summary>
+    /// Values of the object represented as a list of nested <see cref="GraphQLObjectField"/> nodes.
+    /// </summary>
     public List<GraphQLObjectField>? Fields { get; set; }
+
+    /// <summary>
+    /// Object value represented as <see cref="IDictionary{TKey, TValue}"/>.
+    /// <br/>
+    /// This property allocates the string on the heap on first access
+    /// and then caches it as long as <see cref="Value"/> does not change.
+    /// </summary>
+    public IDictionary<string, object?> TypedValue
+    {
+        get
+        {
+            if (Fields == null || Fields.Count == 0)
+                return _empty;
+
+            if (_object == null)
+            {
+                var @object = new Dictionary<string, object?>(Fields.Count);
+                foreach (var field in Fields)
+                    @object.Add(field.Name.StringValue, field.Value.ClrValue);
+                _object = @object;
+            }
+
+            return _object;
+        }
+    }
+
+    /// <inheritdoc />
+    public override object? ClrValue => TypedValue;
```

Comment:
```suggestion
    public override object? ClrValue => _object ??= TypedValue;
```
(needs other changes here)

And similar changes throughout, so it inlines.

Reply:
done